qmake/builddir.pri

```diff
@@ -0,0 +1,34 @@
+# Copyright 2005-2018 The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+# Globally use "/debug" or "/release"
+# as build directory, if it's not
+# a shadow build, otherwise use
+# the shadowed directory.
+#
+# https://wiki.qt.io/QMake-top-level-srcdir-and-builddir
+
+defined(TOP_BUILDDIR, var) {
+	CONFIG *= custom_build_dir
+
+	# If the build and source paths are equal,
+	# it's not a shadow build.
+	equals(TOP_BUILDDIR, $$TOP_SRCDIR) {
+		CONFIG(debug, debug|release) {
+			DESTDIR = $$TOP_BUILDDIR/debug
+		}
+
+		CONFIG(release, debug|release) {
+			DESTDIR = $$TOP_BUILDDIR/release
+		}
+	} else {
+		DESTDIR = $$TOP_BUILDDIR
+	}
+
+	OBJECTS_DIR = $$DESTDIR/.obj
```

Comment:
How about using OBJECTS_ROOT, MOC_ROOT, RCC_ROOT, UI_ROOT here? Instead over overwriting them in each .pro file. I think this would be more sensible.